benchmark/scripts/benchmark_dpo_loss.py

```diff
@@ -0,0 +1,229 @@
+            in_features=H, out_features=V, bias=bias, dtype=dtype
+        )
+        self.dpo_loss = HF_DPO_Loss(beta=beta, ignore_index=ignore_index)
+
+    def forward(self, x, target):
+        return self.dpo_loss.get_batch_loss_metrics(
+            x,
+            self.lin.weight,
+            target,
+            self.lin.bias if hasattr(self.lin, "bias") else None,
+        )
+
+
+class LigerDPOLoss(torch.nn.Module):
+    def __init__(
+        self,
+        H: int,
+        V: int,
+        dtype: torch.dtype,
+        beta: float = 0.1,
+        ignore_index: int = -100,
+        bias: bool = False,
+    ):
+        super().__init__()
+        self.lin = torch.nn.Linear(
+            in_features=H, out_features=V, bias=bias, dtype=dtype
+        )
+        self.beta = beta
+        self.ignore_index = ignore_index
+
+    def forward(self, x, target):
+        return LigerFusedLinearDPOFunction.apply(
+            x,
+            self.lin.weight,
+            target,
+            self.lin.bias if hasattr(self.lin, "bias") else None,
+            self.ignore_index,
+            self.beta,
+            True,
+        )
+
+
+def bench_memory_dpo_loss(input: SingleBenchmarkRunInput) -> SingleBenchmarkRunOutput:
+    B = input.x // 2
+    T = input.extra_benchmark_config["T"]
+    H = input.extra_benchmark_config["H"]
+    V = input.extra_benchmark_config["V"]
+    dtype = input.extra_benchmark_config["dtype"]
+    bias = input.extra_benchmark_config["bias"]
+    beta = input.extra_benchmark_config["beta"]
+    ignore_index = input.extra_benchmark_config["ignore_index"]
+    provider = input.kernel_provider
+
+    device = "cuda"
+    torch_dpo_loss = TorchDPOLoss(
+        H=H, V=V, dtype=dtype, beta=beta, ignore_index=ignore_index, bias=bias
+    ).to(device)
+    liger_dpo_loss = LigerDPOLoss(
+        H=H, V=V, dtype=dtype, beta=beta, ignore_index=ignore_index, bias=bias
+    ).to(device)
+
+    # Input shape: [B*2, T, H]
+    _input = torch.randn(B * 2, T, H, requires_grad=True, dtype=dtype, device=device)
+    # Target shape: [B*2, T]
+    target = torch.randint(V, (B * 2, T), dtype=torch.long, device=device)
+
+    # Add ignore_index tokens to simulate padding
+    num_elements_to_assign = torch.randint(1, B * 2 * T // 2, (1,)).item()
+    indices_to_assign = torch.randperm(B * 2 * T)[:num_elements_to_assign]
+    target.view(-1)[indices_to_assign] = ignore_index
+
+    def fwd():
+        if provider == "liger":
+            return liger_dpo_loss(_input, target)
+        elif provider == "huggingface":
+            return torch_dpo_loss(_input, target)
+
+    def full():
+        y = fwd()
+        y.backward()
+
+    mem_50, mem_20, mem_80 = _test_memory(full, _iter=10, quantiles=QUANTILES)
+    return SingleBenchmarkRunOutput(
+        y_20=mem_20,
+        y_50=mem_50,
+        y_80=mem_80,
+    )
+
+
+def bench_speed_dpo_loss(input: SingleBenchmarkRunInput) -> SingleBenchmarkRunOutput:
+    B = input.x // 2
```

Comment:
TO show that we have input shape of pairs of preference dataset? (Perhaps this is unnecessary)